File: dotnet/FinkiRasporedi/Models/Base/Professor.cs
using FinkiRasporedi.Models.Enums;
using System.ComponentModel.DataAnnotations;

namespace FinkiRasporedi.Models.Base
{
    public class Professor
    {
        [Key]
        public String Id { get; set; }
        public String Name { get; set; }
        public String? Email { get; set; }
        public ProfessorTitle? ProfessorTitle { get; set; }
    }
}

File: fluttter/simple_app/lib/domain/models/professor.dart
import 'enums/professor_title.dart';

class Professor {
  final String id;
  final String name;
  final String email;
  final ProfessorTitle professorTitle;

  Professor({
    required this.id,
    required this.name,
    required this.email,
    required this.professorTitle,
  });

  factory Professor.fromJson(Map<String, dynamic> json) {
    return Professor(
      id: json['id'],
      name: json['name'],
      email: json['email'],
      professorTitle: ProfessorTitle.values[json['professorTitle']],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'email': email,
      'professorTitle': ProfessorTitle.values.indexOf(professorTitle),
    };
  }
}
